pinot-query-planner/src/main/java/org/apache/pinot/query/planner/logical/RelToPlanNodeConverter.java

```diff
@@ -178,12 +180,15 @@ private static PlanNode convertLogicalJoin(LogicalJoin node, int currentStageId)
 
     // Parse out all equality JOIN conditions
     JoinInfo joinInfo = node.analyzeCondition();
-    FieldSelectionKeySelector leftFieldSelectionKeySelector = new FieldSelectionKeySelector(joinInfo.leftKeys);
-    FieldSelectionKeySelector rightFieldSelectionKeySelector = new FieldSelectionKeySelector(joinInfo.rightKeys);
+    JoinNode.JoinKeys joinKeys = new JoinNode.JoinKeys(new FieldSelectionKeySelector(joinInfo.leftKeys),
+        new FieldSelectionKeySelector(joinInfo.rightKeys));
+    List<RexExpression> joinClause =
+        joinInfo.nonEquiConditions.stream().map(RexExpression::toRexExpression).collect(Collectors.toList());
+    boolean isColocatedJoin =
```

Comment:
this is not going to be generic enough. but we can refactor later. 
(for example 32 partition join 16 partition on 4 servers)

Reply:
To support this, we need to enhance the `WorkerManager` instead of handling it here. Currently it requires both tables to have the same `numPartitions`.

Reply:
yeah it has to be handled by WorkerManager, but here we need to ingest more info besides the boolean flag. e.g. numPartitions on both side, etc.